src/main/java/com/digitalasset/testing/ledger/DamlScriptRunner.java

```diff
@@ -73,23 +70,15 @@ public Builder useWallclockTime(boolean useWallclockTime) {
       return this;
     }
 
-    public Builder projectRoot(Path projectRoot) {
-      try {
-        if (Files.list(projectRoot).noneMatch(damlYamlP()))
-          throw new IllegalArgumentException("Project root must contain a daml.yaml");
+    public Builder damlRoot(Path damlRoot) {
+      if (!isDamlRoot(damlRoot))
+        throw new IllegalArgumentException("DAML root must contain a daml.yaml");
 
-        this.projectRoot = projectRoot;
-        return this;
-      } catch (IOException e) {
-        throw new RuntimeException(e);
-      }
+      this.damlRoot = damlRoot.toFile();
+      return this;
     }
 
     public DamlScriptRunner build() {
-      if (projectRoot == null) {
-        projectRoot = findDamlYaml(darPath.toAbsolutePath().getParent());
-      }
-
```

Comment:
More like replaced with a simpler form. If `damlRoot` is given, that it is used as is, if not the current working directory will be used (as before).

**;TL;DR**
I think the main goal of the changes introduced in #40 and mentioned in #41 was to have the ability to specify where `daml.yaml` is located. This remains, however in a simpler and cleaner form: we no longer try to find its location from a given path, but it is required from the client to specify it's location. I think this is far easier to implement, easier to understand from a client perspective.

Reply:
This is an important change in the semantics of what I have done in the other PR. Is there an agreement what do we prefer:

- If there is no `daml.yaml` in or in one of the parent directories of `projectRoot` (or cwd by default), we fall back the default daml install.
  - Alternative to the above is to print some warning when no daml.yaml found.
- If there is no explicit `daml.yaml` in the above search path, we throw an error.

I'm honestly not a big fan of the default configuration behavior of the SDK (that is, using whatever the default settings are for the currently globally used SDK version is).

Reply:
I do agree that a more explicit approach is better. I wanted to avoid searching for `daml.yaml` as I think that is not necessary. I think a simple common ground here is:
- a `com.digitalasset.testing.junit4.Sandbox` cannot be created unless the `damlRoot` points to a folder where is a `daml.yaml`
- current working directory is used as default
- override is possible via `com.digitalasset.testing.junit4.Sandbox.SandboxBuilder#damlRoot`